packages/embark-library-manager/src/npm.js

```diff
@@ -41,28 +70,37 @@ class Npm {
         this._logger,
         packageName,
         version,
-        'Downloading and installing {{packageName}} {{version}}...',
-        'Still downloading and installing {{packageName}} {{version}}... ({{duration}})',
-        'Finished downloading and installing {{packageName}} {{version}} in {{duration}}',
+        'Downloading and installing {{packageName}} {{version}}...\n',
+        'Still downloading and installing {{packageName}} {{version}}... ({{duration}})\n',
+        'Finished downloading and installing {{packageName}} {{version}} in {{duration}}\n',
         {
           showSpinner: !this._useDashboard,
           interval: this._useDashboard ? 2000 : 1000,
-          longRunningThreshold: 10000
+          longRunningThreshold: 600000
         }
       );
       timer.start();
 
-      // do the package download/install
-      pluginManager.install(packageName, version).then((result) => {
-        timer.end();
-        this._installing[packageName + version].forEach((cb) => {
-          cb(null, result.location);
+      // do the package install
+      const installPath = path.dirname(path.dirname(packagePath));
+      mkdirpSync(installPath);
+      exec(`npm init -y && npm install --no-package-lock --no-save ${packageName}@${version}`, {
+        cwd: installPath
+      }, (err) => {
+        this._installing[packageName + version].forEach(cb => {
+          setImmediate(() => { return err ? cb(err) : cb(null, packagePath); });
```

Comment:
I assume this is for preventing race conditions? If so, can you please add a comment accordingly? It's not clear why `setImmediate()` is needed here otherwise...

Reply:
Two things:

(1) It ensures L95-L103 will run before the callbacks, which is what we need, though I could just rearrange the code to achieve that purpose.

(2) It ensures the callbacks are being fired asynchronously. In various places in embark we fail to do that for `(error, value)` style callbacks and that's really bad because the person passing a callback should be able to assume control flow will move on after the callback is passed and then on another tick of the event loop the code in the callback will run. I wasn't 100% sure if the way this function is being invoked via the event bus already makes it 100% for sure async, so I chose to make sure myself.